### PR TITLE
feat: auth server config provider (AR-1464)

### PR DESCRIPTION
**Pull request by MohamadJaara**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1464" title="AR-1464" target="_blank">TaskAR-1464</a>  API versioning
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

we are missing a way to inject auth server 

### Solutions

add `AuthServerConfigProvider` to inject authServer

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

### Note

atm we are creating a new `AuthenticationScope` so we need to cash it

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt [hunk 1]
```diff
@@ -22,13 +22,15 @@ class WireFirebaseMessagingService : FirebaseMessagingService() {
 
     override fun onNewToken(p0: String) {
         super.onNewToken(p0)
-        coreLogic.getAuthenticationScope().saveNotificationToken(p0, "GCM").let { result ->
+        coreLogic.globalScope {
+            saveNotificationToken(p0, "GCM")
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt:26]: I see that is not your changes, but `p0` and hardcoded value..., I suggest to add some `todo`s to this part of the code.

File: app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt [hunk 2]
```diff
@@ -22,13 +22,15 @@ class WireFirebaseMessagingService : FirebaseMessagingService() {
...
+        }.let { result ->
             when (result) {
                 is SaveNotificationTokenUseCase.Result.Failure.Generic -> {
                     appLogger.e("token registration has an issue : ${result.failure} ")
 
                 }
-                is SaveNotificationTokenUseCase.Result.Success -> {
+                SaveNotificationTokenUseCase.Result.Success -> {
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt:33]: is there any reason to remove this `is` and leave the other one?

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 3]
```diff
@@ -125,15 +129,18 @@ class WireActivityViewModel @Inject constructor(
...
     }
 
-    private fun loadServerConfig(url: String) = runBlocking {
+    private fun loadServerConfig(url: String): ServerConfig.Links? = runBlocking {
         return@runBlocking when (val result = getServerConfigUseCase(url)) {
-            is GetServerConfigResult.Success -> result.serverConfig
-            else -> ServerConfig.DEFAULT
+            is GetServerConfigResult.Success -> result.serverConfig.links
+            // TODO: show error message on failure
+            is GetServerConfigResult.Failure.Generic -> null
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:136]: I suggest to log these cases to with appLogger to be aware of them!

**Review by mchenani — COMMENTED** [review]
well done buddy, added some comments and questions.

File: app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt (hunk 2 above)
Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt:33]: `Success` is an object not a class, so it is not really needed I can add it if you want

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-658/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-658/4/) **failed**.

File: app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt (hunk 2 above)
Comment (mchenani) [reply app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt:33]: I missed that, thanks

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-658/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-658/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3623135)(3623135)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3623135)(3623135)-dev-release.apk)

**Pull request merged**